### DISCUSSION TITLE
PQXDH / Quantum Support?

### DISCUSSION THREAD

**Discussion by sterlingwes** [AsamK/signal-cli#1327] [General]
Curious if the CLI was built with an android version that has the new protocol supported or if there's a way to assess when it landed / in which version?

https://signal.org/blog/pqxdh/

**Comment by AsamK** [comment]
It should be available since the implementation of the kyber pre key store: https://github.com/AsamK/signal-cli/commits/master/lib/src/main/java/org/asamk/signal/manager/storage/prekeys/KyberPreKeyStore.java

Though I've never tested it if it actually works yet.

**Comment by dreirund** [comment]
It seems that signal will enforce PQXDH in the near future:

* https://support.signal.org/hc/en-us/articles/8997185514138-Re-connect-your-primary-device-to-continue-using-Signal-Desktop (at the bottom),
* Signal desktop with version 7.48 starts to show a warning that a re-link should be done soon (and refers to the above URL).
